code/go/0chain.net/chaincore/chain/entity.go

```diff
@@ -611,6 +618,69 @@ func (c *Chain) setupInitialState(initStates *state.InitStates) util.MerklePatri
 	return pmt
 }
 
+func (c *Chain) addInitialStakes(mpt *util.MerklePatriciaTrie, stakes []state.InitStake) error {
+	var edbDelegatePools []event.DelegatePool
+	for _, v := range stakes {
+		key := v.ProviderType + ":stakepool:" + v.ProviderID
+		sp := stakepool.StakePool{}
+		if err := mpt.GetNodeValue(util.Path(encryption.Hash(key)), &sp); err != nil {
+			if err != util.ErrValueNotPresent {
+				continue
```

Comment:
Please remember that 'node not found' error should always be reported, so let's return error instead of ignoring it.

Reply:
I think the context here is a bit different. stakePool is with a certain key can exist or it can be a new stakepool to be inserted, both are expected in this case (since we are dealing with initial stake pools). The error that I'm skipping is anything except `node not found`, because i'm not sure whether I should report it back as an error, or just print as log. I'm open to suggestion about it.

Reply:
Yes, it's a bit different, but still we need to panic here instead of continue or logging only, it is the one of the most dangerous behaviors to skip a 'node not found' error. So please do 
```go
if state.ErrInvalidState(err) {
  logging.Logger.Panic("could not initialize staking", zap.Err(err))
}
```